inspire_mitmproxy/services/base_service.py

```diff
@@ -66,7 +71,7 @@ def _raise_do_not_intercept_if_recording(self, request):
 
     def process_request(self, request):
         try:
-            matched_interaction = self._get_first_matching_interaction(request)
+            matched_interaction = self._get_matching_interaction(request)
         except ScenarioNotInService:
```

Comment:
Maybe add an indication in the error message that the scenario wasn't necessarily just not found, but maybe it was found but max_replay were exceeded?